### PR TITLE
Rename Fsp to FinancialServiceProvider: classes, files and folders

### PR DESCRIPTION
**Pull request by diderikvw**
[AB#27846](https://dev.azure.com/redcrossnl/7d17f2a9-3f18-40f1-8c29-98568fbca7e4/_workitems/edit/27846)

## Describe your changes

Refactoring: according to naming guidelines, use full names and not abbreviations.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added tests wherever relevant
- [x] I have made sure that all automated checks pass before requesting a review
- [ ] I have communicated above whether I need any deviation from our PR guidelines (eg. "I don't want the reviewer to merge on my behalf because...")

### CODE REVIEW

**Review by aberonni — APPROVED** [review]
I've added a comment with a concern related to this change. I'll let you decide whether you would like to merge as-is, or whether you would like to make changes as a consequence. I'm fine with either.

File: interfaces/Portal/src/app/enums/fsp-name.enum.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
-import { FspName as FspNameEnum } from '../../../../../services/121-service/src/fsp/enum/fsp-name.enum';
+import { FinancialServiceProviderName as FinancialServiceProviderNameEnum } from '../../../../../services/121-service/src/financial-service-providers/enum/financial-service-provider-name.enum';
 
 /**
  * Note: Names-definition explicitly used from back-end, to prevent repetition.
  */
-const FspName = FspNameEnum;
-type FspName = FspNameEnum;
+const FspName = FinancialServiceProviderNameEnum;
```
[the remark this answers is not in the document]
Comment (diderikvw) [reply interfaces/Portal/src/app/enums/fsp-name.enum.ts:6]: @aberonni  No, there is not. IMO it should become "FinancialServiceProvider" everywhere it is now "Fsp". However, I want to keep this refactoring PR manageable. A while ago I tried to rename too much at once, which became a complete mess, which I all had to throw away and then do it again step-by-step.

For this renaming aspect, as a next PR I would like to work on renaming "FspQuestion" stuff, also related to renaming the entity (and hence creating a migration), also involving aligning mixed terminology of "Fsp Question" and "Fsp Attribute", which are conceptually the same.

And for "all the other places" where Fsp is used, IMO refactoring/renaming that would also be part of when touching that code there when working on a PBI.

But maybe you see a more optimal approach at refactoring, in general, or for this part.

[the remark this answers is not in the document]
Comment (aberonni) [reply interfaces/Portal/src/app/enums/fsp-name.enum.ts:6]: Actually, no, this makes sense to me. I'd say you can go ahead and merge. The only slight adjustment I would suggest for future PRs would be to add this extra context in the description for the reviewer (ie. something along the lines of "this is only the first chunk of the refactoring work")

**Comment by diderikvw** [comment]
Hi @aberonni , this branch needs a re-base, which GH cannot do automatically. Can you help me to do that? Maybe we can write that checklist while we are at it for everyone.

**Comment by aberonni** [comment]
Hey @diderikvw today I wanted to record a mini lecture on git rebase and was looking for a nice, complex example. Your branch would be a perfect candidate :)

Would it work for you if I try recording the lecture, and then we see if that makes you sufficiently autonomous to do this on your own? Part of it would be some "checklist" / git commands.

**Review by RubenGeo — APPROVED** [review]

**Pull request merged**